addons/utm/models/utm_medium.py

```diff
@@ -26,11 +35,25 @@ def create(self, vals_list):
         return super().create(vals_list)
 
     @api.ondelete(at_uninstall=False)
-    def _unlink_except_utm_medium_email(self):
-        utm_medium_email = self.env.ref('utm.utm_medium_email', raise_if_not_found=False)
-        if utm_medium_email and utm_medium_email in self:
-            raise UserError(_(
-                "The UTM medium '%s' cannot be deleted as it is used in some main "
-                "functional flows, such as the recruitment and the mass mailing.",
-                utm_medium_email.name
-            ))
+    def _unlink_except_utm_medium_record(self):
+        for medium in self.REQUIRED_UTM_MEDIUMS_REF:
+            utm_medium = self.env.ref(medium, raise_if_not_found=False)
+            if utm_medium and utm_medium in self:
+                raise UserError(_(
+                    "Oops, you can't delete the Medium '%s'.\n"
+                    "Doing so would be like tearing down a load-bearing wall \u2014 not the best idea.",
+                    utm_medium.name
+                ))
+
+    def _fetch_or_create_website_medium(self):
```

Comment:
Can't me make it generic (based on REQUIRED_UTM_MEDIUMS_REF) ? Like you give the name (website, email, facebook, ...) and it ref or create utm_medium_{name} . That way we can improve other load-bearing wall medium ?

Reply:
I can do that, do you want me to also modify all the other ref calls that uses those mediums in this PR while I am at it?